docs/debugging-testing/testing/unit_testing.md

```diff
@@ -227,23 +188,31 @@ With the following steps, you can write a unit test. This example creates dummy
     }
     ```
 
-1. Stub all external dependencies. Create the following stub in the Mbed OS root directory `UNITTESTS/stubs`:
-
-    **OtherClass_stub.cpp**
-
+    **CMakeLists.txt**
     ```
-    #include "example/OtherClass.h"
-
-    namespace example {
-
-    int OtherClass::otherFunction() {
-        return 0;
-    }
-
-    }
+    set(TEST_NAME myclass-unittest)
+
+    add_executable(${TEST_NAME})
+  
+    target_sources(${TEST_NAME}
+        PRIVATE
+            ${MBED_PATH}/example/tests/unittests/MyClass/MyClass.cpp
+            test_MyClass.cpp
+    )
+    
+    target_link_libraries(${TEST_NAME}
+        PRIVATE
+            mbed-headers
+            mbed-stubs            
+            gmock_main
+    )
+    
+    add_test(NAME "${TEST_NAME}" COMMAND ${TEST_NAME})
+    
+    set_tests_properties(${TEST_NAME} PROPERTIES LABELS "example")
     ```
 
-This example does not use any Mbed OS code, but if your unit tests do, then remember to update header stubs in `UNITTESTS/target_h` and source stubs in `UNITTESTS/stubs` with any missing type or function declarations.
+This example does not use any Mbed OS code, but if your unit tests do, then remember to update header stubs in `UNITTESTS/target_h` and stubs sources in `UNITTESTS/stubs` with any missing type or function declarations.
```

Comment:
```suggestion
This example does not use any Mbed OS code. If your unit tests do, remember to update header stubs in `UNITTESTS/target_h` and stubs sources in `UNITTESTS/stubs` with any missing type or function declarations.
```

Reply:
This sentence is confusing as the cmakelists adds  mbed-headers and stubs. Also, why would the user need to update headers stubs in target_h?